backend/app.py

```diff
@@ -15,11 +16,67 @@
 
 app = Flask(__name__)
 
+# Allow CORS for all origins
+CORS(app)
+
 @app.route('/')
 def home():
     response = supabase.table("Users").select("*").execute()
     print(response)
     return f"Hello World, from Flask!"
 
+# Fetch trips for a given user_id
+@app.route('/trips/<user_id>', methods={'GET'})
+def get_user_trips(user_id):
+    response = supabase.table('Trip').select('*').eq('user_id', user_id).execute()
+
+    if response.data:
+        return jsonify(response.data), 200
+    else:
+        return jsonify({"error": "No trips found for this user"}), 404
+
+@app.route('/delete_trip', methods=['DELETE'])
+def delete_trip():
+    data = request.get_json()
+    trip_id = data.get('trip_id')
+    user_id = data.get('user_id')
+
+    trip = supabase.table('Trip').select('*').eq('trip_id', trip_id).eq('user_id', user_id). execute()
+
+    if trip.data:
+        supabase.table('Trip').delete().eq('trip_id', trip_id).execute()
+        return jsonify({"message": "Trip deleted successfully"}), 200
+    else:
+        return jsonify({"error": "Trip is not found or is not one of the user's trips"}), 404
+
+@app.route('/edit_trip', methods=['PUT'])
+def edit_trip():
+    data = request.get_json()
+    print(f"Received data: {data}")  # Log received data
+
+    trip_id = data.get('trip_id')
+    trip_name = data.get('trip_name')
+    trip_description = data.get('trip_description')
+    start_date = data.get('start_date')
+    time_updated = data.get('time_updated')
+
+    # Update the trip details in the database
+    update_response = (
+        supabase.table('Trip')
+        .update({
+            'trip_name': trip_name,
+            'trip_description': trip_description,
+            'start_date': start_date,
+            'time_updated': time_updated
+        })
+        .eq('trip_id', trip_id)
+        .execute()
+    )
+
+    if update_response.data:
+        return jsonify({"message": "Trip updated successfully"}), 200
+    else:
+        return jsonify({"error": "Trip not found or failed to update"}), 404
+
```

Comment:
I haven't done much research on this myself yet, but is it possible to break up the route logic into controller files to help keep the main app.py file a bit shorter? For example, a TripController.py file that exports the get_user_trips, delete_trip and edit_trip functions? I would imagine this app.py file would get pretty huge as we all start to create the functions for our routes.